app/controllers/concerns/flash_truncation.rb

```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+# Module FlashTruncation provides the truncate_flash method to automatically
+# trim long flash messages to prevent them from overflowing the cookie
+#
+# @author Genome Research Ltd.
+#
+module FlashTruncation
+  # Encoding a json string results in a two-byte overhead for the " either side.
+  # Taking this into account is strictly unnecessary, as we've already got a bit
+  # of overhead built in, but lets keep things as predictable as possible.
+  STRING_OVERHEAD = 2
+
+  #
+  # Truncates the flash message to avoid an ActionDispatch::Cookies::CookieOverflow.
+  # Maximum cookie size is checked against ActionDispatch::Cookies::MAX_COOKIE_SIZE which is 4096
+  # bytes; however:
+  # - This is the size of the session cookie post encryption, which inflates the size
+  # - This cookie also needs to contain other data, such as the session_id, user_uuid and user_name
+  #
+  # @param [Array, String] message The flash to truncate
+  # @param [Integer] max_size The maximum allowed flash size
+  #
+  # @return [Array, String] The truncated message
+  #
+  def truncate_flash(message, max_size = max_flash_size)
+    return message if message.to_json.bytesize <= max_size
+
+    case message
+    when String
+      message.truncate(max_size - STRING_OVERHEAD)
```

Comment:
if this is coming from `truncate_flash_array`, and `max_size` here is 2 or below, this will output '...'
Will this be a problem?

Reply:
You're right. I've specifically built in some overhead to allow for edge cases like this. I did consider handling this case, but considered:
1) If we reduced the size of '...' in response, it becomes increasingly unclear that the message has been truncated. If max size was 2, we'd just end up with an empty string, which doesn't communicate anything at all.
2) It seemed an unnecessary increase in code complexity, especially if we tried to also solve problem 1.

Reply:
Fair enough. It was making my brain hurt slightly. Where did you build in the extra overhead?